rosbag2_compression/include/rosbag2_compression/sequential_compression_reader.hpp

```diff
@@ -70,22 +70,18 @@ class ROSBAG2_COMPRESSION_PUBLIC SequentialCompressionReader
 
 protected:
   /**
-   * Increment the current file iterator to point to the next file in the list of relative file
-   * paths.
-   *
-   * Expected usage:
-   * if (has_next_file()) load_next_file();
+   * Decompress the current bagfile so that it can be opened by the storage implementation.
    */
-  void load_next_file() override;
+  void preprocess_current_file() override;
 
+private:
   /**
    * Initializes the decompressor if a compression mode is specified in the metadata.
    *
    * \throws std::invalid_argument If compression format doesn't exist.
```

Comment:
It also can throw `std::runtime_error` if it couldn't initialize the decompressor